.editorconfig

```diff
@@ -0,0 +1,18 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.neon]
+indent_size = 2
+
+[*.xml.dist]
+indent_size = 2
```

Comment:
did you base this off of what the current file was or did you just make it?

Reply:
No. This is based on how to easily autoformat the package files on IDE. It is used as a baseline so that PHPStorm or Vscode or any IDE can easily format the file on saving or on editing. Most PHP libraries comes with those to have a unify coding practice/style.

Reply:
I know what `.editorconfig` is, I was just asking how you came up with the values for the configuration.

Reply:
The value are the on I use ... the default is set for PHP according to PHP standard coding style from [PSR-1](https://www.php-fig.org/psr/psr-1/)

The other also stem form standard practices in PHP world

Reply:
Ah, makes sense.